### DISCUSSION TITLE
【续 #227】关于修改核心附加安装项的讨论

### DISCUSSION THREAD

**Discussion by Pigeon0v0** [Meloong-Git/PCL#4220] [讨论]
接续 #227 讨论，制作该功能（修改已有核心的附加安装项）前，存在一些问题需要解决，例如：

- 如果在之前安装核心时，使用了默认的名称（如 1.20.1-Forge_xx.x.xx），是否要更改这个名称以配合新的安装项？

- 如果更换了 Mod 加载器，如何处理已有 mods 等文件夹内的内容？

- 其他更多...

欢迎发表自己的意见，或是提出问题

**Comment by zackzheng1121** [comment]
我的想法很簡單，當同時安裝forge和optfine 的時候將optfine作為模組安裝

**Reply by Pigeon0v0** [reply]
重点是当 Forge 换成 Fabric 怎么整

**Reply by zackzheng1121** [reply]
> 重点是当 Forge 换成 Fabric 怎么整

這個問題我不知道

**Reply by wuliaodexiaoluo** [reply]
要不然跟 Mod 管理一样把后缀全部改成 disabled？或者给个提示？

**Reply by Deep-Dark-Forest** [reply]
> 重点是当 Forge 换成 Fabric 怎么整

自动安装 optifabric？如果没有可用版本就弹出警告

**Reply by zackzheng1121** [reply]
> > 重点是当 Forge 换成 Fabric 怎么整
> 
> 自动安装 optifabric？如果没有可用版本就弹出警告

好提議

**Reply by Pigeon0v0** [reply]
> 自动安装 optifabric？如果没有可用版本就弹出警告

其他 Mod 呢

**Reply by Deep-Dark-Forest** [reply]
> > 自动安装 optifabric？如果没有可用版本就弹出警告
> 
> 其他 Mod 呢

可以在 modrinth 和 curseforge 上搜索有没有对应的 Fabric 版本，如果有就下载并替换，如果没有就弹出警告
当然这一切 （包括optifine问题） 都应该是可选的，用户应该可以自行选择是否对模组进行 “Fabric化”

**Reply by zackzheng1121** [reply]
> > > 自动安装 optifabric？如果没有可用版本就弹出警告
> > 
> > 其他 Mod 呢
> 
> 可以在 modrinth 和 curseforge 上搜索有没有对应的 Fabric 版本，如果有就下载并替换，如果没有就弹出警告
> 当然这一切 （包括optifine问题） 都应该是可选的，用户应该可以自行选择是否对模组进行 “Fabric化”

我是覺得可以跳個提示讓用戶選擇是否要自行轉換，當然用戶還是要承擔一定風險

**Reply by Enigma-Soul** [reply]
> > > > 自动安装 optifabric？如果没有可用版本就弹出警告
> > > 
> > > 
> > > 其他 Mod 呢
> > 
> > 
> > 可以在 modrinth 和 curseforge 上搜索有没有对应的 Fabric 版本，如果有就下载并替换，如果没有就弹出警告
> > 当然这一切 （包括optifine问题） 都应该是可选的，用户应该可以自行选择是否对模组进行 “Fabric化”
> 
> 我是覺得可以跳個提示讓用戶選擇是否要自行轉換，當然用戶還是要承擔一定風險

Optfine的兼容性并不强    和“钠”等元素周期表Mod比较  根本不是一个量级  所以应该优先选择“钠”等  这样风险会比之前小很多

**Reply by Deep-Dark-Forest** [reply]
> > > > > 自动安装 optifabric？如果没有可用版本就弹出警告
> > > > 
> > > > 
> > > > 其他 Mod 呢
> > > 
> > > 
> > > 可以在 modrinth 和 curseforge 上搜索有没有对应的 Fabric 版本，如果有就下载并替换，如果没有就弹出警告
> > > 当然这一切 （包括optifine问题） 都应该是可选的，用户应该可以自行选择是否对模组进行 “Fabric化”
> > 
> > 我是覺得可以跳個提示讓用戶選擇是否要自行轉換，當然用戶還是要承擔一定風險
> 
> Optfine的兼容性并不强    和“钠”等元素周期表Mod比较  根本不是一个量级  所以应该优先选择“钠”等  这样风险会比之前小很多

意思是 “Fabric 化” 时用户装了 OptiFine 就会自动换成 Sodium？我觉得这不好，因为用户可能是为了一些目的才选择的 OptiFine（比如自己的 OF 披风，或者自己用的光影不兼容其他模组），如果 Fabric 化以后直接变成 Sodium 了，用户发现自己的 OF 披风也没了光影也用不了了，一脸懵（

**Reply by Enigma-Soul** [reply]
> > > > > > 自动安装 optifabric？如果没有可用版本就弹出警告
> > > > > 
> > > > > 
> > > > > 其他 Mod 呢
> > > > 
> > > > 
> > > > 可以在 modrinth 和 curseforge 上搜索有没有对应的 Fabric 版本，如果有就下载并替换，如果没有就弹出警告
> > > > 当然这一切 （包括optifine问题） 都应该是可选的，用户应该可以自行选择是否对模组进行 “Fabric化”
> > > 
> > > 
> > > 我是覺得可以跳個提示讓用戶選擇是否要自行轉換，當然用戶還是要承擔一定風險
> > 
> > 
> > Optfine的兼容性并不强    和“钠”等元素周期表Mod比较  根本不是一个量级  所以应该优先选择“钠”等  这样风险会比之前小很多
> 
> 意思是 “Fabric 化” 时用户装了 OptiFine 就会自动换成 Sodium？我觉得这不好，因为用户可能是为了一些目的才选择的 OptiFine（比如自己的 OF 披风，或者自己用的光影不兼容其他模组），如果 Fabric 化以后直接变成 Sodium 了，用户发现自己的 OF 披风也没了光影也用不了了，一脸懵（

钠扩展能用光影 OF披风另设置

**Reply by zackzheng1121** [reply]
> > > > 自动安装 optifabric？如果没有可用版本就弹出警告
> > > 
> > > 其他 Mod 呢
> > 
> > 可以在 modrinth 和 curseforge 上搜索有没有对应的 Fabric 版本，如果有就下载并替换，如果没有就弹出警告
> > 当然这一切 （包括optifine问题） 都应该是可选的，用户应该可以自行选择是否对模组进行 “Fabric化”
> 
> 我是覺得可以跳個提示讓用戶選擇是否要自行轉換，當然用戶還是要承擔一定風險

Optfine 的話...這個光影模組的bug很多，所以新版本很少在裝

**Reply by zackzheng1121** [reply]
> > > > 自动安装 optifabric？如果没有可用版本就弹出警告
> > > 
> > > 其他 Mod 呢
> > 
> > 可以在 modrinth 和 curseforge 上搜索有没有对应的 Fabric 版本，如果有就下载并替换，如果没有就弹出警告
> > 当然这一切 （包括optifine问题） 都应该是可选的，用户应该可以自行选择是否对模组进行 “Fabric化”
> 
> 我是覺得可以跳個提示讓用戶選擇是否要自行轉換，當然用戶還是要承擔一定風險

Optfine 的話...這個光影模組的bug很多，所以新版本很少在裝

**Reply by Deep-Dark-Forest** [reply]
> > > > > > > 自动安装 optifabric？如果没有可用版本就弹出警告
> > > > > > 
> > > > > > 
> > > > > > 其他 Mod 呢
> > > > > 
> > > > > 
> > > > > 可以在 modrinth 和 curseforge 上搜索有没有对应的 Fabric 版本，如果有就下载并替换，如果没有就弹出警告
> > > > > 当然这一切 （包括optifine问题） 都应该是可选的，用户应该可以自行选择是否对模组进行 “Fabric化”
> > > > 
> > > > 
> > > > 我是覺得可以跳個提示讓用戶選擇是否要自行轉換，當然用戶還是要承擔一定風險
> > > 
> > > 
> > > Optfine的兼容性并不强    和“钠”等元素周期表Mod比较  根本不是一个量级  所以应该优先选择“钠”等  这样风险会比之前小很多
> > 
> > 意思是 “Fabric 化” 时用户装了 OptiFine 就会自动换成 Sodium？我觉得这不好，因为用户可能是为了一些目的才选择的 OptiFine（比如自己的 OF 披风，或者自己用的光影不兼容其他模组），如果 Fabric 化以后直接变成 Sodium 了，用户发现自己的 OF 披风也没了光影也用不了了，一脸懵（
> 
> 钠扩展能用光影 OF披风另设置

……有些光影只支持 optifine 不支持 sodium

**Comment by 3gf8jv4dv** [comment]
> 如果在之前安装核心时，使用了默认的名称（如 1.20.1-Forge_xx.x.xx），是否要更改这个名称以配合新的安装项

How to identify this *default name* does not seem to be a simple task... However, from the perspective of the PCL named instance mechanism, it is still necessary. If not modified, the actual loader of the instance does not match the loader marked in the instance name, which will cause new problems.

> 如果更换了 Mod 加载器，如何处理已有 mods 等文件夹内的内容？

HMCL’s approach is to do nothing. I personally agree more with @wuliaodexiaoluo’s proposal.

**Comment by SALTWOOD** [comment]
我有个简单粗暴的解决方案
核心更改可以加，但是加一个警告说“不对修改后导致的模组无法使用，光影无法使用等情况负责”
毕竟都改核心了，应该不会是萌新等级的
实在不行就加个提示

也就是说，将修改核心认为是一种用户行为，不做任何动作就可以了

**Reply by zackzheng1121** [reply]
> 我有个简单粗暴的解决方案
> 核心更改可以加，但是加一个警告说“不对修改后导致的模组无法使用，光影无法使用等情况负责”
> 毕竟都改核心了，应该不会是萌新等级的
> 实在不行就加个提示
> 
> 也就是说，将修改核心认为是一种用户行为，不做任何动作就可以了

Wc真狠

**Reply by WForst-Breeze** [reply]
支持，如果在修改核心后，为了追求对修改的模组加载器的兼容性而一并修改了 mods 等文件夹的内容，可能反而弄巧成拙，会有更大的概率把游戏搞坏掉，这样就更不好排查问题在哪了。不如启动器就只负责修改，后续的操作由用户自行进行，这样在减少启动器开发工作量的同时也可以最小化该操作为用户可能带来的额外操作

**Reply by wuliaodexiaoluo** [reply]
>可能反而弄巧成拙，会有更大的概率把游戏搞坏掉

虽然不太可能，但确实有些 Mod 兼容性非常好
~一个 Mod 全平台通用的那种？~

**Reply by MoYuan-CN** [reply]
同意这个解决方案
因为想要找到能正常运行的不同平台相同模组是真的麻烦~~干脆就抛给用户去做~~

**Reply by zackzheng1121** [reply]
~~問題都丟給光影就好~~

**Reply by XingHui-XH** [reply]
这个是真的狠，但是是真的快，可以先推出来，然后升级到完全体的事情就后续考虑，或者干脆就这样不管了

**Comment by zackzheng1121** [comment]
啊6

无聊的小洛 ***@***.***> 於 2024年8月17日 週六 14:51 寫道：

**Comment by zackzheng1121** [comment]
這個就是光影問題了
~~關我們屁事~~

深黑森林 ***@***.***> 於 2024年8月17日 週六 15:29 寫道：

**Reply by Deep-Dark-Forest** [reply]
> 這個就是光影問題了
> 
> ~~關我們屁事~~


所以我说他的方案（把of直接换成sodium）不可行，因为用户一开始选择of肯定是有原因的，如果启动器自动换成of用户肯定不高兴（

**Comment by zackzheng1121** [comment]
那就直接換成optFabric

深黑森林 ***@***.***> 於 2024年8月17日 週六 16:42 寫道：

**Comment by Chiloven945** [comment]
题外话，对于 PCL 应该使用 `版本` 而不是 BakaXL 的 `核心`。

**Reply by Pigeon0v0** [reply]
说习惯了（
